akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -885,6 +951,60 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
 
       expectNoMsg(100.millis)
     }
+    "allow nested persist calls" in {
+      val persistentActor = system.actorOf(Props(classOf[NestedPersists], name, testActor))
+      persistentActor ! "a"
+      persistentActor ! "b"
+
+      expectMsg("a")
+      expectMsg("a-outer")
+      expectMsg("a-inner")
+      // and only then process "b"
+      expectMsg("b")
+      expectMsg("b-outer")
+      expectMsg("b-inner")
```

Comment:
we must document this behavior, I guess it is a good behavior, but it was not obvious for me that it is like that